### PR TITLE
campaign selection: open add-ons manager entry

### LINKED ISSUE #8478: Highlight Add-ons in the Campaign Menu
### Describe the desired feature

Wesnoth has a huge amount of high-quality user-made campaigns, but it's not immediately apparent that clicking the "add-ons" button gives you access to these.

I suggest adding a note at the bottom of the mainline campaigns list, explaining that many high-quality, non-canon campaigns are available through the add-on server, and including a button to reach the add-on server.

### PR DESCRIPTION
**Pull request by babaissarkar**
Resolves #8478 by adding an entry to open the Add-ons manager in the campaign list.

Screenshot from 2024-10-13 09-22-59

### CODE REVIEW

File: src/gui/dialogs/campaign_selection.cpp [hunk 1]
```diff
@@ -30,15 +29,119 @@
...
 
 #include <functional>
-#include "utils/irdya_datetime.hpp"
 
 namespace gui2::dialogs
 {
 
 REGISTER_DIALOG(campaign_selection)
 
+void campaign_selection::update_campaign_list()
```
Comment (Vultraz) [inline src/gui/dialogs/campaign_selection.cpp:42]: Could you merge this back into `pre_show` since it's only used once now? Makes the diff cleaner.

**Comment by babaissarkar** [comment]
Duh, accidentally closed this. :(

**Comment by babaissarkar** [comment]
Please see https://github.com/wesnoth/wesnoth/pull/9489.

File: src/gui/dialogs/campaign_selection.cpp (hunk 1 above)
Comment (babaissarkar) [reply src/gui/dialogs/campaign_selection.cpp]: Updated (and accidentally closed). See https://github.com/wesnoth/wesnoth/pull/9489.

**Pull request closed without merging**